Lib/fontTools/pens/ttGlyphPen.py

```diff
@@ -83,21 +83,24 @@ def glyph(self, componentFlags=0x4):
         assert self._isClosed(), "Didn't close last contour."
 
         almost2 = 1.99993896484375  # F2Dot14 0b1.11111111111111
+        eps = .5 / (1 << 14)
+
+        def closeTo2(value):
+            return True if abs(value - 2) <= eps else False
+
         for i, (glyphName, transformation) in enumerate(self.components):
-            if any(s > 2 or s < -2 for s in transformation[0::3]):
+            if any(s > 2 or s < -2 for s in transformation[:3]):
                 # can't have scale >= 2 or scale < -2:
                 tpen = TransformPen(self, transformation)
                 self.glyphSet[glyphName].draw(tpen)
                 self.components.remove((glyphName, transformation))
-            elif any(almost2 < s <= 2 for s in transformation[0::3]):
+            elif any(closeTo2(s) for s in transformation[:3]):
```

Comment:
wouldn't it be better if you check the closeTo2 stuff as the first `if` branch rather than `elif` one?

Reply:
actiually, the two tests (is any float almost 2, and is any float less than -2 or greater than +2) are independent from each other (as in two separate `if` statements following one another, instead of `elif`): in the former case, we want to round a value of +2.0 to `fixedToFloat(0x7FFF, 14)` to try keep the component as such despite the approximation; in the latter case, we want to decompose the whole composite glyph because we can't fit it anymore.

In the case of `(3, 0, 0, 2, 0, 0)`, we really want to decompose, but instead with the current code the first `if` branch matches because there's `any` float that is `_closeTo2`, and the `elif` one is not.